cpp/src/arrow/util/decimal.h

```diff
@@ -135,6 +159,8 @@ ARROW_EXPORT Decimal128 operator*(const Decimal128& left, const Decimal128& righ
 ARROW_EXPORT Decimal128 operator/(const Decimal128& left, const Decimal128& right);
 ARROW_EXPORT Decimal128 operator%(const Decimal128& left, const Decimal128& right);
 
+ARROW_EXPORT std::ostream& operator<<(std::ostream& os, const Decimal128& value);
```

Comment:
I've noted problems trying to export this symbol in the past, it's often easier to make this a `static inline`